packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding_phone_confirmation/blocs/onboarding_phone_bloc.dart

```diff
@@ -66,27 +67,42 @@ class OnboardingPhoneBloc extends $OnboardingPhoneBloc {
   /// The onboarding service used to communicate the user phone number
   final OnboardingService _onboardingService;
 
+  /// Service used to fetch country codes
+  final SearchCountryCodeService _searchCountryCodeService;
+
   /// The navigation bloc used to navigate the user
   final RouterBlocType _navigationBloc;
 
+  /// The initial country code to be presented. Most apps default to the
+  /// US country code.
+  static const _initialCountryCode = '1';
+
   @override
   Stream<bool> _mapToIsLoadingState() => loadingState;
 
   @override
   Stream<ErrorModel> _mapToErrorsState() => errorState.mapToErrorModel();
 
   @override
-  Stream<bool> _mapToShowErrorsState() =>
-      Rx.combineLatest2(countryCode, phoneNumber, (country, phone) => false)
-          .onErrorReturn(true)
-          .share();
-
-  @override
-  Stream<bool> _mapToSubmitPhoneNumberEnabledState() =>
-      showErrors.map((errors) => !errors).skip(1).startWith(false);
+  Stream<bool> _mapToShowErrorsState() => _$submitPhoneNumberEvent
+      .switchMap((event) => Rx.combineLatest2(
+            countryCode,
+            phoneNumber,
+            (country, phone) => true,
+          ).onErrorReturn(true))
+      .share();
 
   @override
-  Stream<CountryCodeModel?> _mapToCountryCodeState() => _$setCountryCodeEvent;
+  Stream<CountryCodeModel?> _mapToCountryCodeState() => Rx.merge([
+        Stream.value(CountryCodeModel.empty())
+            .switchMap(
+                (_) => _searchCountryCodeService.getItems().asResultStream())
+            .whereSuccess()
+            .map((countryCodes) => countryCodes.firstWhere(
+                (countryCode) => countryCode.code == _initialCountryCode,
+                orElse: () => countryCodes.first)),
```

Comment:
I think we do not need to get the entire country code list. We can have the initial code as model, instead of string, which will reduce the code and improve the performance

Reply:
Fixed.